client/src/components/JobInformation/JobInformation.vue

```diff
@@ -41,7 +41,7 @@ const metadataDetail = ref({
 
 function updateJob(newJob) {
     job.value = newJob;
-    if (newJob) {
+    if (newJob && !invocationId.value) {
```

Comment:
Won't that still poll if there's no invocation ? I have to say I don't understand how any of this works, `/api/invocations` should return an array, so how does it not fail on `invocationId.value = invocation.id; ? Actually, I think it fails, I can see the array being return by the api but no workflow invocation id in the job info component.

![Screenshot 2024-10-30 at 11 34 14](https://github.com/user-attachments/assets/5508659a-dbe0-4b46-9606-4cb049402320)

I think what you want to do is init the `invocationId` with undefined, and only call `fetchInvocation` if the ref is undefined. If you get not invocation back, set the invocation id to null. There's no way in which the job exists but not the invocation if there is an invocation, so you need to call this at most once.

Reply:
> No, it will only poll once because once `invocationId` isn't null anymore,

but it might always be null if there's no invocation, right ?

Reply:
Actually yes 🤦 i forgot about running a tool regardless of invocation....